src/dll.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+"""File implements a simple doubly linked list in python."""
+from __future__ import unicode_literals
+
+
+class Node(object):
+    """Class implements Nodes for use in our dll."""
+    def __init__(self, data=None, nxt=None, prev=None):
+        self.data = data
+        self._next = nxt
+        self._prev = prev
+
+
+class DoublyLinkedList(object):
+    """Class implements a simple doubly linked list."""
+    def __init__(self, iterable=None):
+        self.head = None
+        self.tail = None
+        try:
+            if iterable:
+                for item in iterable:
+                    self.push(item)
+        except TypeError:
+            raise TypeError("Please enter an object that is iterable.")
+
+    def push(self, data):
+        """Insert data at the head of the list."""
+        new_node = Node(data)
+        new_node._prev = None
+        new_node._next = self.head
+        if self.head:
+            self.head._prev = new_node._next
+        self.head = new_node
+        if self.tail is None:
+            self.tail = new_node
+
+    def append(self, data):
+        """Inserts data at the tail of the list."""
+        new_node = Node(data)
+        new_node._next = None
+        new_node._prev = self.tail
+        if self.tail:
+            self.tail._next = new_node._prev
+        self.tail = new_node
+        if self.head is None:
```

Comment:
Line 46 is untested per coverage. Looks like you're not testing the case where you append one item onto an empty list. Make sure you cover this test case!